dao.py

```diff
@@ -645,6 +665,17 @@ def check_creds_and_get_session_id_or_none(self, username, password):
         else:
             return None
 
+    def check_creds(self, username, password):
```

Comment:
we should try not to repeat code (this is basically part of check_creds_and_get_session_id_or_none). maybe let's make `check_creds` return a user object, and refactor `check_creds_and_get_session_id_or_none` to call `check_creds` first?

(this could also be postponed to a later PR, but we should make a note).